docs/reference/cloud-init-deploy.md

```diff
@@ -0,0 +1,161 @@
+---
+myst:
+  html_meta:
+    "description lang=en": "Deploy authd at scale with cloud-init."
+---
+
+# Reference snippets for cloud-init provisioning
+
+[Cloud-init](https://cloudinit.readthedocs.io/en/latest/) is an
+industry-standard method for cloud instance initialization. It can also be used
+to provision client machines during Ubuntu installation.
+
+This page provides example snippets, which can be used in your own cloud config
+YAML files to deploy and configure authd on Ubuntu at scale.
+
+## Setup
+
+Define the necessary environmental variables:
+
+```text
+{% set ISSUER_ID = '<your_issuer_id>' %}
+{% set CLIENT_ID = '<your_client_id>' %}
+```
+
+## Installation
+
+Ensure packages are updated:
+
+```yaml
+package_update: true
+package_upgrade: true
+```
+
+Install the authd deb:
+
+```yaml
+apt:
+  sources:
+      source1:
+          source: 'ppa:ubuntu-enterprise-desktop/authd'
+
+packages:
+  - authd
+  - gnome-shell # only needed for GDM login
+  - yaru-theme-gnome-shell # only needed for GDM login
+```
+
+Then install the broker:
+
+
+:::::{tab-set}
+:sync-group: broker
+
+::::{tab-item} Google IAM
+:sync: google
+
+```yaml
+snap:
+ commands:
+   - ['install', 'authd-google']
+```
+
+::::
+
+::::{tab-item} Microsoft Entra ID
+:sync: msentraid
+
+```yaml
+snap:
+ commands:
+   - ['install', 'authd-msentraid']
+```
+
+::::
+:::::
+
+
+```{tip}
+For more information on installing authd and its brokers, read the
+[installation guide](howto::install).
+```
+
+## Configuration
+
+Configure authd and the broker, ensuring that you edit the allowed suffixes,
+and restart the services for the changes to take effect.
+
+:::::{tab-set}
+:sync-group: broker
+
+::::{tab-item} Google IAM
+:sync: google
+
+```yaml
+write_files:
+  - path: /etc/ssh/sshd_config.d/authd.conf
+    content: |
+      UsePAM yes
+      KbdInteractiveAuthentication yes
```

Comment:
Maybe here we can suggest that they can use

```
Match *@domain.ext
    UsePAM yes
    KbdInteractiveAuthentication yes
```

To ensure that the configuration applies only to the specific user

Reply:
Yes, wildcards are supported, see https://manpages.ubuntu.com/manpages/questing/en/man5/ssh_config.5.html#patterns

Reply:
and I would prefer to stick to `example.com`, which is a reserved domain intended for documentation, which is ensured to not cause naming conflicts when its copied verbatim.

Reply:
this works:

```
Match User *@example.com
    KbdInteractiveAuthentication yes
```

and if the global settings have the Debian default values

```
KbdInteractiveAuthentication no
UsePAM yes
```

Then it does still effectively prevent users who do not match the `*@example` pattern from logging in with a password.

Reply:
Here I would just use

```
Match User *@example.com
    KbdInteractiveAuthentication yes
```

Users can then choose themselves if they want to deviate from the default config and use `PasswordAuthentication no` to disable password authentication for non-authd users. I added a section about that in the security overview.

Reply:
Mmh maybe we should enforce `UsePAM yes` to ensure that login via authd works. So that snippet for `/etc/ssh/sshd_config.d/authd.conf` would become:

```
UsePAM yes
Match User *@example.com
    KbdInteractiveAuthentication yes
```

Do you have an opinion on that, @3v1n0?

Reply:
> Mmh maybe we should enforce UsePAM yes to ensure that login via authd works. 

Eh, I now feel like this is too much bike shedding. Both with or without`UsePAM yes` in the config snippet is fine for me, just go with whatever you prefer Shane (from my side)